examples/pydantic_ai_examples/weather_agent_gradio.py

```diff
@@ -10,18 +10,39 @@
 
 try:
     import gradio as gr
+    from gradio_client import utils as gradio_utils
 except ImportError as e:
     raise ImportError(
         'Please install gradio with `pip install gradio`. You must use python>=3.10.'
     ) from e
 
+# Monkey patch to fix Gradio's JSON schema parser for boolean additionalProperties
+_original_json_schema_to_python_type = gradio_utils._json_schema_to_python_type
```

Comment:
Should this be fixed in Gradio instead?

If using a `BaseModel` fixes it, we don't need this here right?